subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultLenientConfiguration.java

```diff
@@ -335,6 +326,7 @@ public void visitFile(ComponentArtifactIdentifier artifactIdentifier, AttributeC
         }
 
         public Set<File> getFiles() {
+            addArtifactsFiles();
```

Comment:
❌ `addArtifactFiles()` should check whether it already ran, e.g. by having an `artifactFiles` list, which it clears when called the first time. Otherwise mutliple calls to `getFiles()` will give wrong results.